agent/agent_endpoint_test.go

```diff
@@ -2009,9 +2023,11 @@ func TestAgent_Members_ACLFilter(t *testing.T) {
 	testrpc.WaitForLeader(t, a.RPC, "dc1")
 	testrpc.WaitForLeader(t, b.RPC, "dc1")
 
-	joinPath := fmt.Sprintf("/v1/agent/join/127.0.0.1:%d?token=root", b.Config.SerfPortLAN)
+	joinPath := fmt.Sprintf("/v1/agent/join/127.0.0.1:%d", b.Config.SerfPortLAN)
+	req, _ := http.NewRequest("PUT", joinPath, nil)
```

Comment:
nit: The original request was using `httptest.NewRequest` instead of `http.NewRequest` as is done now. Just wondering if there is a reason to use the latter in this case instead?

Reply:
no particular reason, I actually will reverse to `httptest` so I don't accidentally modify the behaviour. Good catch!